### PR TITLE
Random seek while playing audio file

### PR DESCRIPTION
**Pull request by meshonline**
Hi,

My project uses a huge audio file to play sounds, and I want to play only certain pieces of the audio file.

So I achieved a random seek function to jump to any time while playing, and it supports all audio file types except buffered stream, for buffered stream usually feeds data on the fly.

Usage:
// Jump to 12.5 seconds from beginning.
musicSource->Seek(12.5f);

Regards,

Art

### CODE REVIEW

**Comment by cadaver** [comment]
Two things, please mind the coding convention regarding on which lines braces are inserted, and script bindings. Otherwise looking good.

File: Source/Urho3D/Audio/SoundStream.h [hunk 1]
```diff
@@ -36,6 +36,9 @@ class URHO3D_API SoundStream : public RefCounted
     /// Destruct.
     ~SoundStream();
 
+    /// Seek to sample number
+    virtual int Seek(unsigned int sample_number) const { return 1; }
```
Comment (eugeneko) [inline Source/Urho3D/Audio/SoundStream.h]: Please comment return value.

Why 1?

Comment (meshonline) [inline Source/Urho3D/Audio/SoundStream.h]: Fixed it, should work now.

**Comment by eugeneko** [comment]
I still don't understand meaning of Seek return value.
BTW, why do you need `#include "../Audio/OggVorbisSoundStream.h"`?

**Comment by meshonline** [comment]
It does not need to include "../Audio/OggVorbisSoundStream.h", I will remove the line later.
The return value is C style like that in the STB module, I will change to C++ style later too.

**Comment by meshonline** [comment]
I have never tried script binding before, please help.

**Comment by cadaver** [comment]
The short of it is that for Angelscript, you add the function definitions to AudioAPI.cpp to the relevant place (there is a function per class that registers bindings) and for Lua, you add the new function to SoundSource.pkg in Source/Urho3D/LuaScript/pkgs/Audio. You pretty much learn by looking at the existing functions' bindings.

**Comment by meshonline** [comment]
AudioAPI.cpp has not export useful SoundSource interface, so I will not touch the file, which means that we can not use the method in Angel script, I only added the method to Lua bindings.

**Comment by cadaver** [comment]
In this case SoundSource AS registration uses helper functions in APITemplates.h to register both classes' base interface, look there and you'll find it does expose things, and Seek would already go there.

**Comment by meshonline** [comment]
Yes, I see, finally I add the method into AudioAPI.cpp.
Thanks you, I have learned a lot today.

**Comment by cadaver** [comment]
Merged as squashed. Thanks!

**Pull request closed without merging**